spark-cassandra-connector-java/src/main/java/com/datastax/spark/connector/japi/RDDJavaFunctions.java

```diff
@@ -0,0 +1,129 @@
+import com.datastax.spark.connector.writer.WriteConf;
+
+
+/**
+ * A Java API wrapper over {@link RDD} to provide Spark Cassandra Connector functionality.
+ * <p/>
+ * To obtain an instance of this wrapper, use one of the factory methods in {@link CassandraJavaUtil}
+ * class.
+ */
+public class RDDJavaFunctions<T> extends RDDAndDStreamCommonJavaFunctions<T> {
+    public final RDD<T> rdd;
+    public final RDDFunctions<T> rddFunctions;
+
+    public RDDJavaFunctions(RDD<T> rdd) {
+        this.rdd = rdd;
+        this.rddFunctions = new RDDFunctions<>(rdd);
+    }
+
+    public CassandraConnector defaultConnector() {
+        return CassandraConnector$.MODULE$.apply(rdd.conf());
+    }
+
+    public SparkConf getConf() {
+        return rdd.conf();
+    }
+
+    public void saveToCassandra(String keyspace,
+                                String table,
+                                RowWriterFactory<T> rowWriterFactory,
+                                ColumnSelector columnNames,
+                                WriteConf conf,
+                                CassandraConnector connector) {
+
+        rddFunctions.saveToCassandra(keyspace, table, columnNames, conf, connector, rowWriterFactory);
+    }
+
+    /**
+     * Applies a function to each item, and groups consecutive items having the same value together.
+     * Contrary to {@code groupBy}, items from the same group must be already next to each other in the
+     * original collection. Works locally on each partition, so items from different partitions will
+     * never be placed in the same group.
+     */
+    public <U> JavaPairRDD<U, Iterable<T>> spanBy(final Function<T, U> f, ClassTag<U> keyClassTag) {
+        ClassTag<Tuple2<U, Iterable<T>>> tupleClassTag = classTag(Tuple2.class);
+        ClassTag<Iterable<T>> iterableClassTag = CassandraJavaUtil.classTag(Iterable.class);
+
+        RDD<Tuple2<U, Iterable<T>>> newRDD = rddFunctions.spanBy(toScalaFunction1(f))
+                .map(JavaApiHelper.<U, T, scala.collection.Iterable<T>>valuesAsJavaIterable(), tupleClassTag);
+
+        return new JavaPairRDD<>(newRDD, keyClassTag, iterableClassTag);
+    }
+
+    /**
+     * Uses the data from {@code RDD} to join with a Cassandra table without retrieving the entire table.
+     * Any RDD which can be used to saveToCassandra can be used to joinWithCassandra as well as any RDD
+     * which only specifies the partition Key of a Cassandra Table. This method executes single partition
+     * requests against the Cassandra Table and accepts the functional modifiers that a normal
+     * {@link CassandraTableScanRDD} takes.
+     * <p/>
+     * By default this method only uses the Partition Key for joining but any combination of columns
+     * which are acceptable to C* can be used in the join. Specify columns using joinColumns as a
+     * parameter or the {@code on()} method.
+     */
+    public <R> CassandraJavaPairRDD<T, R> joinWithCassandraTable(
```

Comment:
Style, One param per line here
I think with invocation we break it up as well See Spark-99